### PR TITLE
fix extra disk removal

### PR DESCRIPTION
**Pull request by SimonPe**
I have no idea how that `libvirt_domain.name + "-" + disk[:device] + ".raw"` was suposed to work, but the disk gets created using `disk[:path]` so use that for destruction too

### CODE REVIEW

File: lib/vagrant-libvirt/action/destroy_domain.rb [hunk 1]
```diff
@@ -37,9 +37,7 @@ def call(env)
 
             env[:machine].provider_config.disks.each do |disk|
               next if disk[:allow_existing] # shared disks remove only manualy or ???
-              diskname = libvirt_domain.name + "-" + disk[:device] + ".raw"
-              # diskname is uniq
-              env[:libvirt_compute].volumes.all.select{|x| x.name == diskname }.first.destroy
+              env[:libvirt_compute].volumes.all.select{|x| x.name == disk[:path] }.first.destroy
```
Comment (pronix) [inline lib/vagrant-libvirt/action/destroy_domain.rb]: ```disk[:path]``` for my configuration is nil

**Comment by pronix** [comment]
check https://github.com/pradels/vagrant-libvirt/commit/182809e7213cefb985c4db0cfe76c3ef79e4ce24
should be helpful for you.

**Comment by SimonPe** [comment]
@pronix: oh, i missed an `disk[:path] ||= _disk_name(@name, disk)` when reading `create_domain.rb` adding that here to my patch should fix it, will do so and rebase.

**Comment by SimonPe** [comment]
@pronix : fixed it now, care to have a look?

**Comment by pronix** [comment]
can you show vagrantfile where realy require this logic ?

**Comment by SimonPe** [comment]
@pronix: when you manually set the `disk[:path]`, which is allowed by the current api, you also have to use that path when removing the disk again, this is a regression since 0.0.25.
take a look at `create_domain.rb` where the disk creation happens.

**Comment by SimonPe** [comment]
@pronix : could you please motivate that close? The issue still persists, and the fix is rather trivial, so what is the reason you close this?

**Comment by pronix** [comment]
can you show vagrantfile where realy require this logic ?

**Comment by SimonPe** [comment]
@pronix : 
```ruby
Vagrant.configure("2") do |config|
  config.vm.provider :libvirt do |libvirt|
    libvirt.storage :file, :size => '40G', :path => 'datadisk.img'
  end
end
```
`:path` is documented in `README.md` on line 307

File: lib/vagrant-libvirt/action/destroy_domain.rb [hunk 2]
```diff
@@ -39,10 +39,10 @@ def call(env)
...
               # shared disks remove only manualy or ???
               next if disk[:allow_existing]
-              diskname = libvirt_domain.name + '-' + disk[:device] + '.' + disk[:type].to_s
+              disk[:path] ||= "#{libvirt_domain.name}-#{disk[:device]}.#{disk[:type]}"
               # diskname is uniq
               libvirt_disk = env[:libvirt_compute].volumes.all.select do |x|
-                x.name == diskname
+                x.name == disk[:path]
               end.first
               libvirt_disk.destroy if libvirt_disk
```
Comment (pronix) [inline lib/vagrant-libvirt/action/destroy_domain.rb]: you can implement use path as second option if ```libvirt_disk``` if null in this string.

**Comment by SimonPe** [comment]
> you can implement use path as second option if libvirt_disk if null in this string.

what I do in the patch is mirroring the logic used to create the disk in the first place. Which is what is needed here

if you change it to your sugestion, and i have by accident an image in my pool wich matches `"#{libvirt_domain.name}-#{disk[:device]}.#{disk[:type]}"` (which could be the reason I have to set `:path` manualy) then with your sugestion you will end up deleting the wrong disk.

**Comment by pronix** [comment]
sorry, i have problems with long phrases.
1. attemp find disk by patern  "#{libvirt_domain.name}-#{disk[:device]}.#{disk[:type]}"
2. if not found - use path.

**Comment by SimonPe** [comment]
> 1. attemp find disk by patern "#{libvirt_domain.name}-#{disk[:device]}.#{disk[:type]}"
> 2. if not found - use path.

this is the wrong way around, don't you get it?
look at `create_domain.rb` on line 77.

**Comment by pronix** [comment]
destroy working in destroy_domain.

**Comment by pronix** [comment]
destroy working in destroy_domain.

**Comment by SimonPe** [comment]
> destroy working in destroy_domain.

I have no idea how to parse that, could you rephrase?

**Comment by pronix** [comment]
1. attemp find disk by patern "#{libvirt_domain.name}-#{disk[:device]}.#{disk[:type]}"
2. if not found - use path.

**Comment by SimonPe** [comment]
> 1. attemp find disk by patern "#{libvirt_domain.name}-#{disk[:device]}.#{disk[:type]}"
> 2. if not found - use path.

like I already told you if you look at `create_domain.rb` line 77, you will see that this is the wrong way.

the correct way (as implemented by the patch) is:

1. try `disk[:path]`
2. if `disk[:path]` is `nil` use `"#{libvirt_domain.name}-#{disk[:device]}.#{disk[:type]}"`

**Comment by SimonPe** [comment]
@pradels @sciurus could any of you take a look at this.

**Comment by howleyt** [comment]
@SimonPe @pronix What's the current state of this patch?

**Comment by pronix** [comment]
patch is crap but issue is fixed.

**Comment by SimonPe** [comment]
@pronix @howleyt : issue still exists and suposed 'fix' is wrong.

**Comment by SimonPe** [comment]
@pronix @howleyt patch is still valid, and suposed 'fix' is wrong
